### DISCUSSION TITLE
Support for multiple object stores assigned to a set of Databases

### DISCUSSION THREAD

**Discussion by ganderaj** [milvus-io/milvus#26243] [Q&A and General discussion] [answered]
We are implementing Milvus on AWS EKS for enterprise usage at our organization. Our end users would span across different Business Units.

I write to check with the group on the application feasibility to support multiple S3 Buckets (and Prefixes) dedicated to corresponding Business Units. 

[Helm YAML File](https://[raw.githubusercontent.com/milvus-io/milvus-helm/master/charts/milvus/values.yaml](https://raw.githubusercontent.com/milvus-io/milvus-helm/master/charts/milvus/values.yaml)); does not talk about providing multiple such S3 Buckets. Is there any Milvus Operator which accomplishes the ask ?

**Comment by LoveEachDay** [comment, marked as answer]
We don't  support writing data to multiple s3 buckets.

**Reply by ganderaj** [reply]
Thank you for your inputs.

We are under the impression that AWS S3 has its own limitation when it comes to Network bandwidth for GET/PUT object operations. Can you suggest a solution to reduce load on S3 should we are enable 15,000+ users work concurrently with our Milvus DB on AWS EKS (with single S3 as object store) ?

**Reply by yhmo** [reply]
You can build an upper application that has a connection pool and balance the requests from 15000+ users.

**Comment by xiaofan-luan** [comment]
> We are implementing Milvus on AWS EKS for enterprise usage at our organization. Our end users would span across different Business Units.
> 
> I write to check with the group on the application feasibility to support multiple S3 Buckets (and Prefixes) dedicated to corresponding Business Units.
> 
> [Helm YAML File](https://%5Braw.githubusercontent.com/milvus-io/milvus-helm/master/charts/milvus/values.yaml%5D(https://raw.githubusercontent.com/milvus-io/milvus-helm/master/charts/milvus/values.yaml)); does not talk about providing multiple such S3 Buckets. Is there any Milvus Operator which accomplishes the ask ?

Did you already hit the rate limit of S3?
I ask this because milvus only write S3 when flush or compaction and read S3 on load/failure recovery.
I don't see any specific reason why S3 can reach the limit unless you do manual flush every write or frequent deletes.
